contracts/zero-ex/contracts/test/foundry/utils/ForkUtils.sol

```diff
@@ -0,0 +1,336 @@
+import "@0x/contracts-erc20/contracts/src/v06/IEtherTokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "src/transformers/bridges/BridgeProtocols.sol";
+import "src/transformers/bridges/EthereumBridgeAdapter.sol";
+import "src/transformers/bridges/PolygonBridgeAdapter.sol";
+import "src/transformers/bridges/BSCBridgeAdapter.sol";
+import "src/transformers/bridges/ArbitrumBridgeAdapter.sol";
+import "src/transformers/bridges/OptimismBridgeAdapter.sol";
+import "src/transformers/bridges/AvalancheBridgeAdapter.sol";
+import "src/transformers/bridges/FantomBridgeAdapter.sol";
+import "src/transformers/bridges/CeloBridgeAdapter.sol";
+import "src/IZeroEx.sol";
+
+//contract-addresses/addresses.json interfaces
+//need to be alphebetized in solidity but not in addresses.json
+struct Addresses {
+    address affiliateFeeTransformer;
+    address erc20BridgeProxy;
+    address erc20BridgeSampler;
+    address etherToken;
+    address payable exchangeProxy;
+    address exchangeProxyFlashWallet;
+    address exchangeProxyGovernor;
+    address exchangeProxyLiquidityProviderSandbox;
+    address exchangeProxyTransformerDeployer;
+    address fillQuoteTransformer;
+    address payTakerTransformer;
+    address positiveSlippageFeeTransformer;
+    address staking;
+    address stakingProxy;
+    address wethTransformer;
+    address zeroExGovernor;
+    address zrxToken;
+    address zrxTreasury;
+    address zrxVault;
+}
+
+struct TokenAddresses {
+    IERC20TokenV06 DAI;
+    IERC20TokenV06 USDC;
+    IERC20TokenV06 USDT;
+    IEtherTokenV06 WrappedNativeToken;
+}
+
+struct LiquiditySources {
+    address UniswapV2Router;
+}
+
+interface IFQT {
+    function bridgeAdapter() external returns (address);
+}
+
+interface IUniswapV2Router01 {
+    function getAmountsOut(uint256 amountIn, address[] calldata path) external view returns (uint256[] memory amounts);
+
+    function getAmountsIn(uint256 amountOut, address[] calldata path) external view returns (uint256[] memory amounts);
+}
+
+contract ForkUtils is Test {
+    using stdJson for string;
+
+    string json;
+
+    //forked providers for each chain
+    mapping(string => uint256) public forkIds;
+
+    IZeroEx IZERO_EX;
+
+    IBridgeAdapter bridgeAdapter;
+    FillQuoteTransformer fillQuoteTransformer;
+
+    TokenAddresses tokens;
+    Addresses addresses;
+    LiquiditySources sources;
+
+    uint256 forkBlock = 15_000_000;
+
+    string[] chains = ["mainnet", "bsc", "polygon", "avalanche", "fantom", "optimism", "arbitrum"];
+    string[] indexChainIds = [".1", ".56", ".137", ".43114", ".250", ".10", ".42161"];
+    string[] chainIds = ["1", "56", "137", "43114", "250", "10", "42161"];
+    uint256[] chainId = [1, 56, 137, 43114, 250, 10, 42161];
+
+    //special fork block number for fantom since it produces blocks faster and more frequently
+    uint256[] blockNumber = [forkBlock, forkBlock, 33447149, forkBlock, 32000000, forkBlock, forkBlock];
```

Comment:
Wouldn't this break if we were to no longer use a node that has Archive support (e.g internal nodes)?

Reply:
i think it would, but we can modify it to get the current block number from the RPC isntead of giving it a specific `forkBlock`

Reply:
It seems like only `fantom` and `polygon` need a special `blockNumber`, but we force every chain to use a specific `blockNumber`, its a bit non obvious and would likely be a footgun in the future. Does everyone know these forked tests all run at  block `15_000_000`? What if someone is writing a forked test for a newly released liquidity source?

Reply:
Is this so that caching makes the tests run faster?